File: mycollab-ui/src/main/scala/com/mycollab/vaadin/events/PreviewFormHandler.scala
package com.mycollab.vaadin.events

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
trait PreviewFormHandler[T] {
  /**
    *
    * @param data
    */
  def gotoNext(data: T)

  /**
    * @param data
    */
  def gotoPrevious(data: T)

  /**
    * @param data
    */
  def onAssign(data: T)

  /**
    * @param data
    */
  def onEdit(data: T)

  /**
    *
    * @param data
    */
  def onAdd(data: T)

  /**
    *
    * @param data
    */
  def onDelete(data: T)

  def onPrint(source: Object, data: T)

  /**
    * @param data
    */
  def onClone(data: T)

  /**
    *
    */
  def onCancel()

  /**
    * @param action
    * @param data
    */
  def onExtraAction(action: String, data: T): Unit
}

File: mycollab-ui/src/main/java/com/mycollab/vaadin/events/PreviewFormHandler.kt
package com.mycollab.vaadin.events

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
interface PreviewFormHandler<T> {
    /**
     *
     * @param data
     */
    fun gotoNext(data: T)

    /**
     * @param data
     */
    fun gotoPrevious(data: T)

    /**
     * @param data
     */
    fun onAssign(data: T)

    /**
     * @param data
     */
    fun onEdit(data: T)

    /**
     *
     * @param data
     */
    fun onAdd(data: T)

    /**
     *
     * @param data
     */
    fun onDelete(data: T)

    fun onPrint(source: Object, data: T)

    /**
     * @param data
     */
    fun onClone(data: T)

    /**
     *
     */
    fun onCancel()

    /**
     * @param action
     * @param data
     */
    fun onExtraAction(action: String, data: T)
}